pkg/engine/executor/dataobjscan.go

```diff
@@ -278,15 +282,15 @@ func (s *dataobjScan) getLessFunc(dir physical.Direction) func(a, b logs.Record)
 		return labels.Compare(aStream, bStream) < 0
 	}
 
-	switch dir {
-	case physical.Forward:
+	switch direction {
+	case physical.ASC:
 		return func(a, b logs.Record) bool {
 			if a.Timestamp.Equal(b.Timestamp) {
 				compareStreams(a, b)
 			}
 			return a.Timestamp.After(b.Timestamp)
```

Comment:
should this not be using Before for ASC ts ordering?

Reply:
I was confused about this as well, but it's a how the Heap is implemented.
If you want least-ascending, then you need to compare `a > b`, see https://github.com/grafana/loki/blob/main/pkg/util/topk/topk_test.go#L33-L50 (note that the slice is reversed after popping the greatest/smallest elements).